concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+            )
+        )
+
+
+def check_error_message_value_is_different_than_payment_value(
+    errors: List[Error],
+    matched_objects: List,
+    force_payment_committed_pending_response: PendingResponse,
+) -> None:
+    """
+    Add error if PaymentInfo.amount_paid related to stored PendingResponse of type ForcePaymentCommitted
+    differs from the settlement_payment.amount value.
+    """
+    payment_info = force_payment_committed_pending_response.payments.order_by('id').last()
+    for settlement_payment in matched_objects:
+        if settlement_payment.amount != payment_info.amount_paid:
+            errors.append(
+                Error(
+                    matching_error=MatchingError.MESSAGE_VALUE_DIFFERS_FROM_PAYMENT_VALUE,
+                    message=f'PaymentInfo with pk {payment_info.pk} amount_paid value {payment_info.amount_paid} '
+                            f'related to stored PendingResponse of type ForcePaymentCommitted '
+                            f'with pk {force_payment_committed_pending_response.pk} '
+                            f'differs from the settlement payment with tx_hash {settlement_payment.tx_hash} '
+                            f'amount value {settlement_payment.amount}.',
+                )
+            )
+
+
+def check_error_message_payment_ts_is_different_than_payment_closure_time(
+    errors: List[Error],
+    matched_objects: List,
+    force_payment_committed_pending_response: PendingResponse,
+) -> None:
+    """
+    Add error if PaymentInfo.payment_ts related stored PendingResponse of type ForcePaymentCommitted
+    differs from closure_time of the settlement payment.
+    """
+    payment_info = force_payment_committed_pending_response.payments.order_by('id').last()
+    for settlement_payment in matched_objects:
+        if settlement_payment.closure_time != parse_datetime_to_timestamp(payment_info.payment_ts):
+            errors.append(
+                Error(
+                    matching_error=MatchingError.FORCE_PAYMENT_COMMITED_PAYMENT_TS_DIFFERS_FROM_SETTLEMENT_PAYMENT_CLOSURE_TIMESTAMP,
+                    message=f'PaymentInfo with pk {payment_info.pk} payment ts value {payment_info.payment_ts} '
+                            f'related to stored PendingResponse of type ForcePaymentCommitted '
+                            f'with pk {force_payment_committed_pending_response.pk} '
+                            f'differs from the settlement payment with tx_hash {settlement_payment.tx_hash} '
+                            f'closure time value {settlement_payment.closure_time}.',
+                )
+            )
+
+
+def check_error_more_than_one_subtask_result_of_the_same_type_for_subtask(
+    errors: List[Error],
+    matched_objects: List,
+    additional_verification_payment: CoverAdditionalVerificationEvent,
+) -> None:
+    """ Add error if there is more than two SubtaskResultsSettled or SubtaskResultsRejected for the same subtask. """
```

Comment:
Your `if` is not checking for more than two `SubtaskResultsSettled` or `SubtaskResultsRejected`. Only for more than one.

Maybe you mean `if there exist both SubtaskResultsSettled and SubtaskResultsRejected for the same subtask`?